### PR TITLE
Remove JSON schemas in favour of structs

### PR DESCRIPTION
**Pull request by Mrtenz**
This replaces the `package.json` and `snap.manifest.json` JSON schemas in favour of `superstruct`.

## To do

- [x] Fix browser export
- [x] Fix other packages

### CODE REVIEW

File: packages/utils/src/structs.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { refine, Struct } from 'superstruct';
+
+/**
+ * Validate that a value has a minimum length.
+ *
+ * @param struct - The struct to validate.
+ * @param min - The minimum length.
+ * @returns The struct.
+ */
+export function minSize(struct: Struct<string>, min: number) {
```
Comment (ritave) [inline packages/utils/src/structs.ts:10]: Already exists in superstruct https://docs.superstructjs.org/api-reference/refinements#size

File: packages/utils/src/types.ts [hunk 2]
```diff
@@ -3,7 +3,168 @@ import {
...
+  optional,
+  pattern,
+  record,
+  refine,
+  size,
+  string,
+  type,
+  union,
+} from 'superstruct';
+import { valid } from 'semver';
```
Comment (ritave) [inline packages/utils/src/types.ts:22]:
```suggestion
import { valid as validSemver } from 'semver';
```

File: packages/utils/src/types.ts [hunk 3]
```diff
@@ -3,7 +3,168 @@ import {
...
+  1,
+  214,
+);
+
+// Note we use `type` instead of `object` here, because the latter does not
+// allow unknown keys.
+export const NpmSnapPackageJsonStruct = type({
+  version: VersionStruct,
+  name: NameStruct,
+  private: optional(boolean()),
```
Comment (ritave) [inline packages/utils/src/types.ts:52]: I'm pretty sure we don't use some of those fields

**Review by ritave — CHANGES_REQUESTED** [review]
I'm generally all for this change.

Could we talk through implementing / not implementing  https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-4.md before merging this though? No need to duplicate efforts

File: packages/utils/src/structs.ts (hunk 1 above)
Comment (Mrtenz) [reply packages/utils/src/structs.ts:10]: `size(string(), 1)` is a string with a size of exactly 1, not a minimum size unfortunately.

File: packages/utils/src/types.ts (hunk 3 above)
Comment (Mrtenz) [reply packages/utils/src/types.ts]: Removed them. We can add them again in the future if necessary.

File: packages/utils/src/structs.ts (hunk 1 above)
Comment (ritave) [reply packages/utils/src/structs.ts:10]: Would `size(string(), 1, Infinity)` work?

Comment (Mrtenz) [reply packages/utils/src/structs.ts]: Ah, I missed that bit. Fixed!

**Review by ritave — DISMISSED** [review]

File: packages/utils/package.json [hunk 4]
```diff
@@ -60,7 +56,6 @@
     "@metamask/utils": "^3.1.0",
     "@noble/hashes": "^1.1.3",
     "@scure/base": "^1.1.1",
-    "ajv": "^8.11.0",
```
Comment (FrederikBolding) [inline packages/utils/package.json:63]: Do we still need `ajv-cli`?

File: packages/utils/src/npm.ts [hunk 5]
```diff
@@ -70,13 +65,9 @@ export function validateNpmSnap(
   }
 
   try {
-    validateSnapJsonFile(NpmSnapFileNames.PackageJson, packageJson);
+    assertIsNpmSnapPackageJson(packageJson);
   } catch (error) {
-    throw new Error(
-      `${errorPrefix ?? ''}"${NpmSnapFileNames.PackageJson}" is invalid:\n${
```
Comment (FrederikBolding) [inline packages/utils/src/npm.ts:76]: Do we not want the filenames to part of the error message anymore?

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**